src/Kunstmaan/SeoBundle/Helper/Menu/SeoManagementMenuAdaptor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Kunstmaan\SeoBundle\Helper\Menu;
+
+use Kunstmaan\AdminBundle\Helper\Menu\MenuItem;
+use Kunstmaan\AdminBundle\Helper\Menu\MenuBuilder;
+use Kunstmaan\AdminBundle\Helper\Menu\MenuAdaptorInterface;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\HttpFoundation\Request;
+
+class SeoManagementMenuAdaptor implements MenuAdaptorInterface
+{
+    /** @var  Container */
+    private $container;
+
+    /**
+     * @param Container $container
+     */
+    public function __construct(Container $container)
+    {
+        $this->container        = $container;
+    }
+
+    /**
+     * In this method you can add children for a specific parent, but also remove and change the already created children
+     *
+     * @param MenuBuilder $menu The MenuBuilder
+     * @param MenuItem[] &$children The current children
+     * @param MenuItem $parent The parent Menu item
+     * @param Request $request The Request
+     */
+    public function adaptChildren(MenuBuilder $menu, array &$children, MenuItem $parent = null, Request $request = null)
+    {
+        if (!is_null($parent)) {
+            if ('KunstmaanAdminBundle_settings' == $parent->getRoute()) {
+                if ($this->container->get('security.context')->isGranted('ROLE_SUPER_ADMIN')) {
```

Comment:
This if condition can be added to the one above, since you don't do anything else inside ...